### ISSUE TITLE
MySQL Dialect: BOOLEAN to BIT(1) instead of TINYINT(1)

### ISSUE THREAD

**Issue by mdemblani** [sequelize/sequelize#10288]
This ticket is not an issue rather a question and furthermore a proposal for a future update, if it can be done. MYSQL as a Database doesnot explicitly support a BOOLEAN but rather supports two different field types: `TinyInt` and `Bit`. Out of this, many of the exisitng ORM's including Sequelize use `TinyInt(1)` instead. 

The issue that `TinyInt(1)` is that apart from supporting only `0` and `1` it also allows you to add values from the range `2 - 9`, which can in-fact lead to unstable data in the database.  A rather suitable option would be to use `Bit(1)` which would only allow you to add `0` or `1` in the database, with only `1` being considered to be **truthy** and everything else **falsy**

**Comment by sushantdhiman** [comment]
There is already an issue reported to `mysql2` https://github.com/sidorares/node-mysql2/issues/350

We can possibly parse buffer value as boolean, but wont this break old installations

**Comment by mdemblani** [comment]
Yes this would lead to a break in the old installations. When this is implemented, an update/support for the old installations would also be needed.
